runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -608,32 +608,37 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
 			sizeRequirements.lineNumberTableSize = U_32(sizeInformation.lineNumberSize);
 			sizeRequirements.localVariableTableSize = U_32(sizeInformation.variableInfoSize);
 
-			if ( sharedStoreClassTransaction.allocateSharedClass(&sizeRequirements) ){
-				U_8 *romClassBuffer = (U_8*)sharedStoreClassTransaction.getRomClass();
-				/*
-				 * Make note that the laydown is occurring in SharedClasses
-				 */
-				romSize = finishPrepareAndLaydown(
-						(U_8*)sharedStoreClassTransaction.getRomClass(),
-						(U_8*)sharedStoreClassTransaction.getLineNumberTable(),
-						(U_8*)sharedStoreClassTransaction.getLocalVariableTable(),
-						&sizeInformation, modifiers, extraModifiers, optionalFlags,
-						true, sharedStoreClassTransaction.hasSharedStringTableLock(),
-						&classFileOracle, &srpOffsetTable, &srpKeyProducer, &romClassWriter,
-						context, &constantPoolMap
-						);
-
-				fixReturnBytecodes(_portLibrary, (J9ROMClass *)romClassBuffer);
+			if (!sharedStoreClassTransaction.isCacheFull()) {
```

Comment:
I added this check because when reaching this point, we do not have write mutex if the cache is full/soft full. Without this check,  j9shr_classStoreTransaction_createSharedClass() does not guarantee returning on checking J9SHR_RUNTIMEFLAG_AVAILABLE_SPACE_FULL from localRuntimeFlags (Another thread that has write mutex may unset this flag), leading to unexpected behavior.